src/test/java/com/exasol/dbbuilder/dialects/postgres/PostgreSqlImmediateDatabaseObjectWriterTest.java

```diff
@@ -0,0 +1,15 @@
+package com.exasol.dbbuilder.dialects.postgres;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.exasol.dbbuilder.dialects.AbstractImmediateDatabaseObjectWriterTest;
+import com.exasol.dbbuilder.dialects.DatabaseObjectWriter;
+
+@ExtendWith(MockitoExtension.class)
+class PostgreSqlImmediateDatabaseObjectWriterTest extends AbstractImmediateDatabaseObjectWriterTest {
+    @Override
+    protected DatabaseObjectWriter getDatabaseObjectWriter() {
+        return new PostgreSqlImmediateDatabaseObjectWriter(this.connectionMock);
+    }
```

Comment:
Shouldn't we add tests for the public interface as well (`public void write(final User user)`, `public void write(final User user, final GlobalPrivilege... privileges`, `public void drop(final Schema schema)`)?

Reply:
I would prefer to have both integration and unit tests, but as you want.